lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -199,30 +199,31 @@ export default class BitcoinProcessor {
     const lastBlockHeight = await this.bitcoinClient.getCurrentBlockHeight();
     const lastBlockInfo = await this.bitcoinClient.getBlockInfoFromHeight(lastBlockHeight);
 
-    const numOfBlocksToProcess = lastBlockHeight - startingBlock.height + 1;
-
     // a map of all blocks mapped with their hash being the key
     const notYetValidatedBlocks: Map<string, IBlockInfoExtended> = new Map();
     // An array of blocks representing the validated chain sorted by height
-    const validatedBlocks: (IBlockInfoExtended | undefined)[] = [...new Array(numOfBlocksToProcess)];
+    const validatedBlocks: IBlockInfoExtended[] = [];
 
     console.log(`Begin fast processing block ${startingBlock.height} to ${lastBlockHeight}`);
     // Loop through files backwards and process blocks from the end/tip of the blockchain until we reach the starting block given.
     let hashOfEarliestKnownValidBlock = lastBlockInfo.hash;
     let heightOfEarliestKnownValidBlock = lastBlockInfo.height;
     while (bitcoinBlockDataIterator.hasPrevious() && heightOfEarliestKnownValidBlock >= startingBlock.height) {
-      const blockData = bitcoinBlockDataIterator.previous()!;
-      await this.processBlockData(blockData, notYetValidatedBlocks, startingBlock.height, heightOfEarliestKnownValidBlock);
-      [heightOfEarliestKnownValidBlock, hashOfEarliestKnownValidBlock] = await this.findEarliestValidBlockAndAddToValidBlocks(
+      const blocks = bitcoinBlockDataIterator.previous()!;
+      await this.processBlocks(blocks, notYetValidatedBlocks, startingBlock.height, heightOfEarliestKnownValidBlock);
+      this.findEarliestValidBlockAndAddToValidBlocks(
         validatedBlocks,
         notYetValidatedBlocks,
-        heightOfEarliestKnownValidBlock,
         hashOfEarliestKnownValidBlock,
         startingBlock.height);
+      if (validatedBlocks.length) {
```

Comment:
Style nit: don't like javascript treating 0 as false.
```suggestion
      if (validatedBlocks.length > 0) {
```